rtl/eclrtl/CMakeLists.txt

```diff
@@ -80,7 +80,8 @@ include_directories (
 ADD_DEFINITIONS( -D_USRDLL -DECLRTL_EXPORTS )
 
 if (USE_BOOST_REGEX)
-    if ((CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANGXX) AND (${BOOST_REGEX_VERSION} LESS 105700) AND NOT ICU_REQUIRES_CPP11)
```

Comment:
Should we remove the ICU_REQUIRES_CPP11 option now - I think this is the only usage of it?

Reply:
Yeah I'll go ahead and remove it since it's unused now

Reply:
@richardkchapman that option has been removed and this PR is ready for merge.